src/tasks/AppleAppBuilder/Templates/runtime-coreclr.m

```diff
@@ -0,0 +1,170 @@
+    path = [path stringByAppendingString:@"/Contents/Resources"];
+#endif
+
+    bundle_path = strdup ([path UTF8String]);
+
+    return bundle_path;
+}
+
+const char *
+get_tpa_list (const char *bundle)
+{
+    if (tpa_list)
+        return tpa_list;
+    NSString *path = @(bundle_path);
+    NSArray *dirs = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:path error:NULL];
+    NSMutableArray *assemblies = [[NSMutableArray alloc] init];
+    [dirs enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+        NSString *filename = (NSString *)obj;
+        NSString *extension = [[filename pathExtension] lowercaseString];
+        if ([extension isEqualToString:@"dll"]) {
+            [assemblies addObject:[path stringByAppendingPathComponent:filename]];
+        }
+    }];
+
+    NSString *ns_tpa_list = [assemblies componentsJoinedByString:@":"];
+    tpa_list = strdup ([ns_tpa_list UTF8String]);
+
+    return tpa_list;
+}
+
+char *
+strdup_printf (const char *msg, ...)
+{
+    va_list args;
+    char *formatted = NULL;
+    va_start (args, msg);
+    vasprintf (&formatted, msg, args);
+    va_end (args);
+    return formatted;
+}
+
+void
+log_callback (const char *log_domain, const char *log_level, const char *message, bool fatal, void *user_data)
+{
+    os_log_info (OS_LOG_DEFAULT, "(%s %s) %s", log_domain, log_level, message);
+    if (fatal) {
+        os_log_info (OS_LOG_DEFAULT, "Exit code: %d.", 1);
+        exit (1);
+    }
+}
+
+void *
+pinvoke_override (const char *libraryName, const char *entrypointName)
+{
+    void *symbol = NULL;
+
+    if (strcmp (libraryName, "__Internal") == 0) {
+        symbol = dlsym (RTLD_DEFAULT, entrypointName);
+    }
+    return symbol;
+}
+
+void
+mono_ios_runtime_init (void)
+{
+#if INVARIANT_GLOBALIZATION
+    setenv ("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", "1", TRUE);
+#endif
+
+    id args_array = [[NSProcessInfo processInfo] arguments];
+    assert ([args_array count] <= 128);
+    const char *managed_argv [128];
+    int argi;
+    for (argi = 0; argi < [args_array count]; argi++) {
+        NSString* arg = [args_array objectAtIndex: argi];
+        managed_argv[argi] = [arg UTF8String];
+    }
+
+    const char* bundle = get_bundle_path ();
+    chdir (bundle);
+
+    const char* tpa = get_tpa_list (bundle);
+
+    char icu_dat_path [1024];
+    int res;
+
+    res = snprintf (icu_dat_path, sizeof (icu_dat_path) - 1, "%s/%s", bundle, "icudt.dat");
+    assert (res > 0);
+
+    // TODO: set TRUSTED_PLATFORM_ASSEMBLIES, APP_PATHS and NATIVE_DLL_SEARCH_DIRECTORIES
+    const char *appctx_keys [] = {
+        "RUNTIME_IDENTIFIER", 
+        "APP_CONTEXT_BASE_DIRECTORY",
+#if !defined(INVARIANT_GLOBALIZATION) && !TARGET_OS_MACCATALYST
+        "ICU_DAT_FILE_PATH"
+#endif
+    };
+    const char *appctx_values [] = {
+        APPLE_RUNTIME_IDENTIFIER,
+        bundle,
+#if !defined(INVARIANT_GLOBALIZATION) && !TARGET_OS_MACCATALYST
+        icu_dat_path
+#endif
+    };
```

Comment:
I don't think it would hurt anything to add these to propertyKeys/propertyValues.  they're literally the same thing.